### PR TITLE
Rewrite respondent-home-ui into Python

### PR DESCRIPTION
**Pull request by jcox-dev**
This replaces the existing ruby application with a Python3/aiohttp rewrite.

### CODE REVIEW

**Comment by jcox-dev** [comment]
[Dev task 7: EXT: Access eQ social survey (RH) (???)](https://trello.com/c/SwkUMf9O/99-dev-task-7-ext-access-eq-social-survey-rh)

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
+[[source]]
+
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+
+[packages]
+
+aiohttp = "*"
```
Comment (tej99) [inline Pipfile]: Should these be pinned to specific versions?

File: app/app.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    app.on_startup.append(on_startup)
+    app.on_cleanup.append(on_cleanup)
+
+    logger.info("App setup complete", config=app_config["ENV"])
+
+    return app
+
+
+if __name__ == "__main__":
+    app = create_app()
```
Comment (tej99) [inline app/app.py]: Why does this get duplicated in app/__main__.py?

File: app/eq.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+            ),
+            "employment_date": find_event_date_by_tag(
+                "employment", self._collex_events, self._collex_id, False
+            ),
+            "return_by": find_event_date_by_tag(
+                "return_by", self._collex_events, self._collex_id, True
+            ),
+        }
+
+    def _get_party_by_business_id(self):
```
Comment (tej99) [inline app/eq.py]: Shouldn't this be a coroutine? async def?

Comment (jcox-dev) [reply app/eq.py]: @tej99 still all wip. Sorry for missing off the labels. We wanted to get the automated builds running for docker, etc.

File: README.md [hunk 4]
```diff
@@ -1,122 +0,0 @@
-# Respondent Home Ruby Web Application
```
Comment (shahi645) [inline README.md]: @jcox-dev Should Update the `readme`?

File: tests/test_eq.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import unittest
+
+from app.eq import format_date
+from app.exceptions import InvalidEqPayLoad
+
+
+class TestGenerateEqURL(unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (jcox-dev) [reply tests/test_eq.py]: @benjefferies Old name. Might be time to split things up anyway.

File: README.md (hunk 4 above)
Comment (jcox-dev) [reply README.md]: Done 👍

File: app/config.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+
+class DevelopmentConfig:
+    env = Env()
+    HOST = env.str("HOST", default="0.0.0.0")
+    PORT = env.int("PORT", default="9092")
+    LOG_LEVEL = env("LOG_LEVEL", default="INFO")
+    STATIC_ROOT = "app/static"
+
+    ACCOUNT_SERVICE_URL = env.str("ACCOUNT_SERVICE_URL", default="http://0.0.0.0:9092")
```
Comment (shahi645) [inline app/config.py]: @jcox-dev I think this is supposed to be localhost? for all of them?

Comment (jcox-dev) [reply app/config.py]: Can be either I think 🤷‍♂️

File: tests/test_eq.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (jcox-dev) [reply tests/test_eq.py]: I've split up the tests in the [latest commit](https://github.com/ONSdigital/respondent-home-ui/pull/22/commits/264b47983f9162281253a903eff3d705d01acffb).

File: app/eq.py [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+
+
+def handle_response(response):
+    try:
+        response.raise_for_status()
+    except ClientError as ex:
+        logger.error("Error in response", url=response.url, status_code=response.status)
+        raise ex
+    else:
+        logger.debug("Successfully connected to service", url=response.url)
```
Comment (shahi645) [inline app/eq.py]: should be `info`?

File: app/static/css/responsive.css [hunk 8]
```diff
@@ -0,0 +1,2 @@
+
+@import "https://cdn.ons.gov.uk/sdc/v1.3.0/css/responsive.css";
```
Comment (shahi645) [inline app/static/css/responsive.css]: 👍

File: README.md [hunk 9]
```diff
@@ -1,122 +1,31 @@
...
 ## Installation
-Install the RubyGems the application depends on by running `bundle install`. Note that this application depends on a private `iac-validator` RubyGem for performing IAC validation. This gem is only used in one place within `routes/home.rb`:
-
-```ruby
-unless InternetAccessCodeValidator.new(iac).valid?
-  ...
-end
-```
+Install the required Python packages for running and testing Respondent Home within a virtual environment:
+  `make install`
```
[the remark this answers is not in the document]
Comment (jcox-dev) [reply README.md]: I'll add those once we're confident they can be used. Adding about the unit tests.

**Pull request merged**